src/machine/machinectl.c

```diff
@@ -2312,7 +2312,12 @@ static int set_limit(int argc, char *argv[], void *userdata) {
         int r;
 
         if (STR_IN_SET(argv[argc-1], "-", "none", "infinity"))
-                limit = (uint64_t) -1;
+                /*
+                 * Linux kernel uses *signed* 64bit numbers for offsets and
+                 * sizes, for example ftruncate() returns EINVAL for length < 0.
+                 * This (8EiB) is VFS limitation for many operations.
+                 */
+                limit = INT64_MAX;
```

Comment:
I don't think this is the right solution. After all, the variable is of type `unit64_t`, and we're passing it to a DBus call as `t`, which translates to 'unsigned 64bit' (as opposed to `x`).

Hence, I think this should remain `-1`, but we need to handle the fact that the kernel takes signed 64bit as argument to `ftruncate()` and friends on the other side, where the call is dispatched.